### PR TITLE
`Lectures`: Polish design and usability of lecture unit management view

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The exercise management view currently looks weird and totally out of line with how the rest of Artemis looks, especially with those rounded buttons and large border radius boxes. Additionally, no drag and drop sort is possible.

grafik

### Description

- Changed the design of this page to look more like all other components we have.
- Grouped all available buttons on the right instead of splitting them between left and right
- Enabled drag and drop sorting
- Moved the "Create new" buttons to the top right as that's like it is everywhere else

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Lecture

1. Open the lecture unit management view of the lecture
2. Add a few units of all types
3. Reorder them with drag and drop and make sure that order is persisted (aka, still there after a reload)
4. Make sure that all buttons still work
5. Make sure that you can still expand them and the content works

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

https://user-images.githubusercontent.com/23171488/183942547-86d47d67-3956-453e-afb6-7fc4ad72064c.mov

### CODE REVIEW

**Review by TheZoker — APPROVED** [review]
Mainly looked at the visual/functional changes on TS3 and everything worked as expected!

**Review by Mikael31 — DISMISSED** [review]
Tested on t2. every button worked and it ran smooth

**Review by stefanwaldhauser — APPROVED** [review]
I am sad to see my old UI gone but you improved it much :D Nice change!

**Comment by b-fein** [comment]
I did not test it, but just looked at the video in the PR description. Is the ‘View Isolated’ button still rounded, or has this been changed since?

Screenshot 2022-08-11 at 07-34-26 `Lectures` Polish design and usability of lecture unit management view by bassner · Pull Request #5595 · ls1intum_Artemis

**Comment by bassner** [comment]
> I did not test it, but just looked at the video in the PR description. Is the ‘View Isolated’ button still rounded, or has this been changed since?
> 
> Screenshot 2022-08-11 at 07-34-26 Lectures Polish design and usability of lecture unit management view by bassner · Pull Request #5595 · ls1intum_Artemis

Yes, we want to change that in a follow up, as well as their position

**Review by Santia-go — APPROVED** [review]
Tested on TS1
Working as expected. 


https://user-images.githubusercontent.com/88991954/184622053-b3495fac-26b5-4744-b261-08fba6a32be4.mov

**Review by ge65cer — APPROVED** [review]
Tested on TS2, found no issues. I'm also good with the code changes. 🙂

File: src/main/webapp/app/overview/course-exercises/course-exercise-row.component.html [hunk 1]
```diff
@@ -37,7 +37,7 @@ <h4 class="fw-medium">{{ exercise.title }}</h4>
                 *ngIf="!isPresentationMode"
             ></jhi-exercise-details-student-actions>
             <div class="exercise-tags col-auto col-sm d-flex justify-content-center mt-2">
-                <h4 class="fw-medium">
+                <h4 class="fw-medium" *ngIf="exercise.releaseDate && dayjs(exercise.releaseDate).isAfter(dayjs())">
```
Comment (Strohgelaender) [inline src/main/webapp/app/overview/course-exercises/course-exercise-row.component.html:40]: Do we need to convert this here? I think this should be done when receiving the data from the server.

Comment (bassner) [reply src/main/webapp/app/overview/course-exercises/course-exercise-row.component.html]: I've tested it and it works without it, too.

**Review by jpbernius — APPROVED** [review]

**Pull request merged**